pkg/controllers/user/rbac/legacy_grb_cleaner.go

```diff
@@ -19,22 +19,28 @@ func newLegacyGRBCleaner(m *manager) *grbCleaner {
 	return &grbCleaner{m: m}
 }
 
+// TODO  Remove this once no longer needed
```

Comment:
Might add a note here saying "and its integration test". Will save some time in the future.

Reply:
This comment should just be dropped, how do you know when it's not needed?

Reply:
It is helpful to acknowledge that its code which is fixing a time-boxed issue though. Maybe re-phrase like _"handles bug that occured from version 2.y-2.z"_.